doc/filters/shuffle.rst

```diff
@@ -0,0 +1,76 @@
+``shuffle``
+========
+
+The ``shuffle`` filter shuffles an array:
+
+.. code-block:: twig
+
+    {% for user in users|shuffle %}
+        ...
+    {% endfor %}
+
+.. caution::
+
+    Internally, Twig uses the PHP `shuffle`_ function.
+    This function assigns new keys to the elements in array. It will remove
+    any existing keys that may have been assigned, rather than just reordering the keys.
```

Comment:
A Twig filter that modifies a PHP data structure passed to the template does not look like a good idea to me.

Reply:
note that it does not modify the data passed to the template though. The shuffled array will be a copy of the original one.

Reply:
If the implementation takes care of it, we should IMO remove this paragraph as it's then irrelevant for users of this filter.

Reply:
Well, what is relevant for the users is that the shuffled array does not preserve keys. So if your input had not sequential keys (where you often don't care about them when iterating) but indexed keys (using the user id for instance), it is not the case anymore after shuffling it.

Reply:
I've added your explanation.
Thank you.